crates/core/src/commands/lock.rs

```diff
@@ -0,0 +1,75 @@
+//! `lock` subcommand
+
+use chrono::{DateTime, Local};
+use log::error;
+use rayon::ThreadPoolBuilder;
+
+use crate::{
+    error::{CommandErrorKind, RepositoryErrorKind, RusticResult},
+    progress::{Progress, ProgressBars},
+    repofile::{configfile::ConfigId, IndexId, KeyId, PackId, RepoId, SnapshotId},
+    repository::Repository,
+    WriteBackend,
+};
+
+pub(super) mod constants {
+    /// The maximum number of reader threads to use for locking.
+    pub(super) const MAX_LOCKER_THREADS_NUM: usize = 20;
+}
+
+pub fn lock_repo<P: ProgressBars, S>(
+    repo: &Repository<P, S>,
+    until: Option<DateTime<Local>>,
+) -> RusticResult<()> {
+    lock_all_files::<P, S, ConfigId>(repo, until)?;
+    lock_all_files::<P, S, KeyId>(repo, until)?;
+    lock_all_files::<P, S, SnapshotId>(repo, until)?;
+    lock_all_files::<P, S, IndexId>(repo, until)?;
+    lock_all_files::<P, S, PackId>(repo, until)?;
+    Ok(())
+}
+
+pub fn lock_all_files<P: ProgressBars, S, ID: RepoId + std::fmt::Debug>(
+    repo: &Repository<P, S>,
+    until: Option<DateTime<Local>>,
+) -> RusticResult<()> {
+    if !repo.be.can_lock() {
+        return Err(CommandErrorKind::NoLockingConfigured.into());
+    }
+
```

Comment:
I see we check that here, but I think this should be checked within the `::lock()` itself, so people using it, can't forget to call it.